olp-cpp-sdk-dataservice-read/tests/SerializerTest.cpp

```diff
@@ -426,4 +429,84 @@ TEST(SerializerTest, VersionResponse) {
   ASSERT_EQ(expectedOutput, json);
 }
 
+TEST(SerializerTest, ConsumerProperties) {
+  std::string expected_output =
+      "{\
+    \"kafkaConsumerProperties\":{\
+    \"key_string\":\"value_string\",\
+    \"key_c-string\":\"value_c-string\",\
+    \"key_int32\":\"42\",\
+    \"key_bool\":\"1\"\
+    }\
+    }";
+
+  olp::dataservice::read::ConsumerOptions expected_options = {
+      olp::dataservice::read::ConsumerOption("key_string",
+                                             std::string("value_string")),
+      olp::dataservice::read::ConsumerOption("key_c-string", "value_c-string"),
+      olp::dataservice::read::ConsumerOption("key_int32", (int32_t)42),
+      olp::dataservice::read::ConsumerOption("key_bool", true)};
+
+  olp::dataservice::read::ConsumerProperties props(expected_options);
+
+  const auto& options = props.GetProperties();
+  ASSERT_EQ(expected_options.size(), options.size());
+  for (size_t i = 0; i < expected_options.size(); ++i) {
+    EXPECT_EQ(expected_options[i].GetKey(), options[i].GetKey());
+    EXPECT_EQ(expected_options[i].GetValue(), options[i].GetValue());
+  }
+
+  auto json = olp::serializer::serialize(props);
+
+  RemoveWhitespaceAndNewlines(expected_output);
+  RemoveWhitespaceAndNewlines(json);
+
+  EXPECT_EQ(expected_output, json);
+}
+
+TEST(SerializerTest, StreamOffset) {
```

Comment:
It's not clear what is this test about. Is it to test the `StreamOffsets ::SetOffsets` method or the `olp::serializer::serialize` method?

Consider splitting it into single-assertion tests or using sub-scopes to cut the test in parts.

Reply:
Good point. I removed the corresponding test about testing `StreamOffsets` methods. I think we need to implement them in separate ticket/PR, because this PR about serializing.